packages/commons/src/auth/authData.ts

```diff
@@ -116,91 +104,97 @@ export type AuthToken = z.infer<typeof AuthToken>;
+  systemRole: undefined;
+  organizationId: TenantId;
+  userId: UserId;
+  userRoles: UserRole[];
+  selfcareId: SelfcareId;
+  externalId: {
+    value: string;
+    origin: string;
+  };
 };
 
-const getUserRoles = (token: AuthToken): UserRole[] =>
-  match(token)
-    .with({ role: "m2m" }, (t) => [t.role])
-    .with({ role: "internal" }, (t) => [t.role])
-    .with({ role: "maintenance" }, (t) => [t.role])
-    .with({ "user-roles": P.not(P.nullish) }, (t) => t["user-roles"])
-    .exhaustive();
+export type M2MAuthData = {
+  systemRole: Extract<SystemRole, "m2m">;
+  organizationId: TenantId;
+};
 
-const getOrganizationId = (token: AuthToken): TenantId | undefined =>
-  match(token)
-    .with({ "user-roles": P.not(P.nullish) }, { role: "m2m" }, (t) =>
-      unsafeBrandId<TenantId>(t.organizationId)
-    )
-    .with({ role: "internal" }, { role: "maintenance" }, () => undefined)
-    .exhaustive();
+export type InternalAuthData = {
+  systemRole: Extract<SystemRole, "internal">;
+};
 
-const getUserId = (token: AuthToken): UserId | undefined =>
-  match(token)
-    .with({ "user-roles": P.not(P.nullish) }, (t) =>
-      unsafeBrandId<UserId>(t.uid)
-    )
-    .with(
-      { role: "m2m" },
-      { role: "internal" },
-      { role: "maintenance" },
-      () => undefined
-    )
-    .exhaustive();
+export type MaintenanceAuthData = {
+  systemRole: Extract<SystemRole, "maintenance">;
+};
 
-const getExternalId = (
-  token: AuthToken
-): { value: string; origin: string } | undefined =>
-  match(token)
-    .with({ "user-roles": P.not(P.nullish) }, (t) => t.externalId)
-    .with(
-      { role: "m2m" },
-      { role: "internal" },
-      { role: "maintenance" },
-      () => undefined
-    )
+export type AuthData =
+  | UIAuthData
+  | M2MAuthData
+  | InternalAuthData
+  | MaintenanceAuthData;
+
+export const getAuthDataFromToken = (token: AuthToken): AuthData =>
+  match<AuthToken, AuthData>(token)
+    .with({ role: "internal" }, () => ({ systemRole: "internal" }))
+    .with({ role: "maintenance" }, () => ({ systemRole: "maintenance" }))
+    .with({ role: "m2m" }, (t) => ({
+      systemRole: "m2m",
+      organizationId: unsafeBrandId<TenantId>(t.organizationId),
+    }))
+    .with({ "user-roles": P.not(P.nullish) }, (t) => ({
+      systemRole: undefined,
+      organizationId: unsafeBrandId<TenantId>(t.organizationId),
+      userId: unsafeBrandId<UserId>(t.uid),
+      userRoles: t["user-roles"],
+      selfcareId: unsafeBrandId<SelfcareId>(t.selfcareId),
+      externalId: t.externalId,
+    }))
     .exhaustive();
 
-const getSelfcareId = (token: AuthToken): SelfcareId | undefined =>
-  match(token)
-    .with({ "user-roles": P.not(P.nullish) }, (t) =>
-      unsafeBrandId<SelfcareId>(t.selfcareId)
-    )
-    .with(
-      { role: "m2m" },
-      { role: "internal" },
-      { role: "maintenance" },
-      () => undefined
-    )
+export function getUserInfoFromAuthData(
+  authData: AuthData | undefined | null
```

Comment:
why using also `null`? it's related to a new possibile value of Authdata generated from JWT?